hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/commit/BaseCommitActionExecutor.java

```diff
@@ -246,6 +246,7 @@ protected HoodieWriteMetadata<HoodieData<WriteStatus>> executeClustering(HoodieC
         .performClustering(clusteringPlan, schema, instantTime);
     HoodieData<WriteStatus> writeStatusList = writeMetadata.getWriteStatuses();
     HoodieData<WriteStatus> statuses = updateIndex(writeStatusList, writeMetadata);
+    context.putCachedDataIds(config.getBasePath(), instantTime, statuses.getId());
```

Comment:
This is quite brittle -- it's far from obvious that we need to persist cached RDD ids somewhere. 
I'd suggest we instead modify `HoodieData.persist` to accept context and this registration internally (so that we can establish it as an invariant that any persisted RDD will be registered)

Reply:
i wasn't happy with tracing every persisting call and thought about this approach but also wanted to keep the impacting scope narrow. A change in all persist() call may lead to unexpected side effects.

Reply:
HoodieData is already tightly coupled (1:1) with HoodieEngineContext so there's nothing shady about HD API accepting HEC.

Current approach doesn't really make sense as it's extremely brittle -- we can't expect that someone will be aware of needing to register the RDD whenever they persist.

Reply:
fixed